### ISSUE TITLE
Integrate Documentation from docs.rs.school into rs.school

### ISSUE THREAD

**Issue by SpaNb4** [rolling-scopes/site#650]
## 📝 Describe the feature

Integrate documentation from https://docs.rs.school into https://rs.school site to improve user experience.

## ℹ️ Additional context

Currently, documentation on docs.rs.school is not easily accessible. This integration will enhance visibility and usability for users.

## ✅ Definition of Done (DoD):

1. Content from `docs.rs.school` is migrated to `rs.school` without loss.
2. Search functionality for documentation is implemented.

**Comment by cootook** [comment]
@SpaNb4 is that issue related to #623 ?

**Comment by SpaNb4** [comment]
> @SpaNb4 is that issue related to #623 ?

No, this issue is not related to #623. I'll update the description shortly
